libs/librrgraph/src/io/rr_graph_reader.cpp

```diff
@@ -115,3 +134,140 @@ void load_rr_file(RRGraphBuilder* rr_graph_builder,
+        hash_combine(s.intra_tile);
+        hash_combine(static_cast<int>(s.type()));
+
+        return seed;
+    }
+};
+
+struct t_rr_switch_inf_equal {
+    bool operator()(const t_rr_switch_inf& lhs, const t_rr_switch_inf& rhs) const {
+        return lhs.R == rhs.R &&
+               lhs.Cin == rhs.Cin &&
+               lhs.Cout == rhs.Cout &&
+               lhs.Cinternal == rhs.Cinternal &&
+               lhs.Tdel == rhs.Tdel &&
+               lhs.mux_trans_size == rhs.mux_trans_size &&
+               lhs.buf_size == rhs.buf_size &&
+               lhs.power_buffer_type == rhs.power_buffer_type &&
+               lhs.power_buffer_size == rhs.power_buffer_size &&
+               lhs.intra_tile == rhs.intra_tile &&
+               lhs.type() == rhs.type();
+    }
+};
+
+void load_rr_edge_overrides(std::string_view filename,
+                            RRGraphBuilder& rr_graph_builder,
+                            const RRGraphView& rr_graph) {
+    std::ifstream file(filename.data());
+    if (!file) {
+        VTR_LOG_ERROR("Failed to open the RR edge override file: %s\n", filename.data());
+    }
+
+    std::unordered_map<t_rr_switch_inf, RRSwitchId, t_rr_switch_inf_hash, t_rr_switch_inf_equal> unique_switch_info;
+    for (const auto& [rr_sw_idx, sw] : rr_graph.rr_switch().pairs()) {
+        unique_switch_info.insert({sw, rr_sw_idx});
+    }
+
+    std::string line;
+    std::vector<float> overridden_values;
+    bool firstLine = true;
+
+    while (std::getline(file, line)) {
+        if (firstLine) {
+            if (line.empty() || line[0] != '#') {
+                VTR_LOG_ERROR("Error: First line must start with #\n");
+            }
+            firstLine = false;
+            continue;  // Ignore first line
+        }
+
+
+        if (!line.empty()) {
+            overridden_values.clear();
+            RREdgeId edge_id = process_rr_edge_override(line, overridden_values, rr_graph);
+            RRSwitchId curr_switch_id = (RRSwitchId)rr_graph.edge_switch(edge_id);
+            t_rr_switch_inf switch_override_info = rr_graph.rr_switch_inf(curr_switch_id);
+
+            switch_override_info.Tdel = (overridden_values.size() >= 1) ? overridden_values[0] : switch_override_info.Tdel;
+            switch_override_info.R = (overridden_values.size() >= 2) ? overridden_values[1] : switch_override_info.R;
+            switch_override_info.Cin = (overridden_values.size() >= 3) ? overridden_values[2] : switch_override_info.Cin;
+            switch_override_info.Cout = (overridden_values.size() >= 4) ? overridden_values[3] : switch_override_info.Cout;
+            switch_override_info.Cinternal = (overridden_values.size() >= 5) ? overridden_values[4] : switch_override_info.Cinternal;
+
+            RRSwitchId new_switch_id;
+            auto it = unique_switch_info.find(switch_override_info);
+            if (it == unique_switch_info.end()) {
+                new_switch_id = rr_graph_builder.add_rr_switch(switch_override_info);
```

Comment:
Now that I think about it, I'm pretty sure I wrote a function that retrieves switch info, searches the RR Node switches for one with similar characteristics, and, if none is found, creates a new one and returns the corresponding RR switch ID (I think the name was something like get_or_create_rr_switch_id). In the issue I mentioned, if you could reference this function, I'll update this part of the code myself.